ReverseIndexManip/AbstractReverseIndexManip.py

```diff
@@ -11,42 +11,99 @@ class abstractReverseIndexManip:
 	Abstract class
 	"""
 
-	def __init__(self, fileName):
+	def __init__(self, fileName, read = False):
 		"""
 		Creates, extracts content, indexes and saves result
+		If read is for read uses otherwise it is for write
 		Abstract
 		"""
 		# Signature -> Distinctive signature in order to check the format
 		self.signature = None
-		# Index file name
-		self.reverseIndex = open(fileName, 'rw')
+		# ReverseIndex file name
+		self.reverseIndex = open(fileName, 'r' if read else 'w')
 		# Current read value
 		self.indexes = []
 		# Documents in order to allow correspondance
 		self.documentsNames = []
 
-	def create(self, indexesFileNameList):
+	def compare(self, key1, key2):
+		"""
+		Compares two keys
+		Returns true if key1 > key2
+		Abstarct
+		"""
+		return key1 > key2
+
+	def selectKey(self):
+		"""
+		Selects the first key to put and returns docs and values
+		Private
+		"""
+		resKey = ''
+		resDocuments = []
+		resValues = []
+		for index, i in zip(self.indexes, range(len(self.indexes))):
+			# There is a key which is not null due to while condition
```

Comment:
Yes this function is a private one which has to be called only if a key is not null, however I will add a comment